ai_ta_backend/main.py

```diff
@@ -373,6 +375,50 @@ def mit_download_course() -> Response:
   response.headers.add('Access-Control-Allow-Origin', '*')
   return response
 
+@app.route('/addCanvasUsers', methods=['GET'])
+def add_canvas_users():
+  """
+  Add users from canvas to the course
+  """
+  print("In /addCanvasUsers")
+
+  canvas = CanvasAPI()
+  canvas_course_id: str = request.args.get('course_id')
+  course_name: str = request.args.get('course_name')
+
+  success_or_failure = canvas.add_users(canvas_course_id, course_name)
+  
+  response = jsonify({"outcome": success_or_failure})
+
+  response.headers.add('Access-Control-Allow-Origin', '*')
+  return response
+
+@app.route('/ingestCanvas', methods=['GET'])
+def ingest_canvas():
+  """
+  Ingest course content from Canvas
+  """
+  canvas = CanvasAPI()
+  canvas_course_id: str = request.args.get('course_id')
+  course_name: str = request.args.get('course_name')
+  success_or_failure = canvas.ingest_course_content(canvas_course_id, course_name)
+  response = jsonify({"outcome": success_or_failure})
+  response.headers.add('Access-Control-Allow-Origin', '*')
+  return response
+
+@app.route('/updateCanvas', methods=['GET'])
+def update_canvas():
+  """
+  Update course content from Canvas
+  """
+  canvas = CanvasAPI()
+  canvas_course_id: str = request.args.get('course_id')
+  course_name: str = request.args.get('course_name')

```

Comment:
Please add parameter checking to these endpoints with the `abort()`. As in `/mit-download`. Thanks.

Reply:
done!